static/js/graphs_page.js

```diff
@@ -1049,17 +1049,61 @@ var graphPage = {
     },
     defaultLayoutWidget: {
         init: function (is_shared) {
-            if (utils.getURLParameter('auto_layout') || _.isNil(is_shared)) {
+            if (_.isNil(is_shared)) {
                 $('#setDefaultLayoutBtn').hide();
+                $('#setDefaultAutoLayoutBtn').hide();
                 $('#removeDefaultLayoutBtn').hide();
-            } else if (utils.getURLParameter('user_layout') && utils.getURLParameter('user_layout') == default_layout_id) {
+            }else if (utils.getURLParameter('auto_layout')) {
                 $('#setDefaultLayoutBtn').hide();
+                $('#setDefaultAutoLayoutBtn').show();
+                $('#removeDefaultLayoutBtn').hide();
+            }else if (utils.getURLParameter('user_layout') && utils.getURLParameter('user_layout') == default_layout_id) {
+                $('#setDefaultLayoutBtn').hide();
+                $('#setDefaultAutoLayoutBtn').hide();
                 $('#removeDefaultLayoutBtn').show();
             } else {
                 $('#setDefaultLayoutBtn').show();
+                $('#setDefaultAutoLayoutBtn').hide();
                 $('#removeDefaultLayoutBtn').hide();
             }
         },
+        onSetDefaultAutoLayoutBtn: function (e) {
+            graphPage.saveLayout(utils.getURLParameter('auto_layout'));
+            $('table').bootstrapTable('refresh')
+
+            apis.layouts.get($('#GraphID').val(), {
```

Comment:
Can we update the method `saveLayout` instead of calling the get layout function again? I think that will be a better approach.

Reply:
@adbharadwaj  I didn't get the exact idea behind this. Can you please elaborate this a bit more?

Reply:
Here is what your pseudocode looks like:
```
saveLayout(layout_name)
layout_id = getLayoutID(layout_name)
setDefaultLayout(graph_id, layout_id)
```
Here is what I would like it to be:
```
layout_id = saveLayout(layout_name)
setDefaultLayout(graph_id, layout_id)
```

Currently, `saveLayout` function doesnt return anything or there is not callback function which accepts layout_id. So inorder to do that you will have add a callback function to saveLayout method.

Reply:
@adbharadwaj I have updated my Pull Request as per your suggestions. Please review it.